### PR TITLE
Update SAMTOOLS_DEPTH

### PR DESCRIPTION
**Pull request by nevinwu**
This PR updates samtools depth module. Now intervals.bed file can be passed as a new input channel and it will  be used as "samtools depth -b intervals.bed". Also, I have updated test data.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: tests/modules/nf-core/samtools/depth/main.nf
Comment (SPPearce) [inline tests/modules/nf-core/samtools/depth/main.nf:1]: Can you add another test that it works if you don't supply the intervals please.

**Review by SPPearce — COMMENTED** [review]
Looks good, can you add a test for the case where you don't provide intervals.

File: tests/modules/nf-core/samtools/depth/main.nf
Comment (nevinwu) [reply tests/modules/nf-core/samtools/depth/main.nf:1]: Done!

**Review by SPPearce — APPROVED** [review]
Perfect, simply done.

**Pull request merged**